.github/workflows/nearcore_deploy.yml

```diff
@@ -0,0 +1,58 @@
+name: Nearcore release
+
+on:
+  push:
+    branches: master
+
+  workflow_dispatch:
+    inputs:
+      branch:
+        default: 'master'
+        description: "Nearcore branch to build and publish"
+        type: string
+        required: true
+
+jobs:
+  binary-release:
+    name: "Build and publish neard binary"
+    runs-on: "ubuntu-22.04-8core"
+    environment: deploy
+    permissions:
+      id-token: write # required to use OIDC authentication
+
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::754641474505:role/GitHubActionsRunner
+          aws-region: us-west-1
+
+      - name: Checkout ${{ github.event.inputs.branch }} branch
+        if: ${{ github.event_name == 'workflow_dispatch'}}
+        uses: actions/checkout@v4
+        with:
+          repository: near/nearcore
+          ref: ${{ github.event.inputs.branch }}
+
+      - name: Checkout nearcore master repository
+        if: ${{ github.event_name != 'workflow_dispatch'}}
+        uses: actions/checkout@v4
+        with:
+          repository: near/nearcore
+          ref: master
+          
+      - name: Log rust and cargo versions 
+        run: |
+          rustup --version
+          cargo --version
```

Comment:
Nit: GHA does quite a good job at showing what the individual steps are if you just keep

```yaml
- run: command
```

without a name, so long as each step is a single command, which in this case should definitely be. My subjective experience is that it makes the workflows easier to understand more maintainable too.

Reply:
That said, these versions are already locked in by the repository contents, what value is there in printing these anymore?